deluge/common.py

```diff
@@ -962,23 +990,51 @@ def is_ipv6(ip):
     """
 
     try:
-        import ipaddress
-    except ImportError:
-        import socket
-
-        try:
-            return socket.inet_pton(socket.AF_INET6, ip)
-        except (OSError, AttributeError):
-            if windows_check():
-                log.warning('Unable to verify IPv6 Address on Windows.')
-                return True
+        socket.inet_pton(socket.AF_INET6, ip)
+    except OSError:
+        return False
     else:
+        return True
+
+
+def is_adapter(adapter):
+    """A test to see if 'interface' is a valid network interface.
+
+    Args:
+        interface (str): The Interface to test. eg. eth0 linux. GUID on Windows.
+
+    Returns:
+        bool: Whether Interface is valid or not.
+
+    Examples:
+        >>> is_adapter("eth0")
+        True
+        >>> is_adapter("{7A30AE62-23ZA-3744-Z844-A5B042524871}")
+        True
+
+    """
+
+    if not windows_check():
         try:
-            return ipaddress.IPv6Address(decode_bytes(ip))
-        except ipaddress.AddressValueError:
+            adapters = socket.if_nameindex()
```

Comment:
Can be simplified `if_nametoindex(adapter)` and replace in else statement any() with True